GChan/ImageLink.cs

```diff
@@ -39,47 +39,55 @@ public string GenerateNewFilename(ImageFileNameFormat format)
 
             string extension = Path.GetExtension(URL); // Contains period (.).
 
-            string result;
-
-            switch (format)
+            string result = format switch
             {
-                case ImageFileNameFormat.ID:
-                    result = lastPart;
-                    break;
-                case ImageFileNameFormat.OriginalFilename:
-                    result = UploadedFilename + extension;
-                    break;
-                case ImageFileNameFormat.IDAndOriginalFilename:
-                default:
-                    result = Path.GetFileNameWithoutExtension(URL) + " - " + UploadedFilename + extension;
-                    break;
-            }
+                ImageFileNameFormat.ID => lastPart,
+                ImageFileNameFormat.OriginalFilename => UploadedFilename + extension,
+                ImageFileNameFormat.IDAndOriginalFilename => $"{Path.GetFileNameWithoutExtension(URL)}-{UploadedFilename}{extension}",
+                _ => $"{UploadedFilename}-{Path.GetFileNameWithoutExtension(URL)}{extension}",

```

Comment:
Since I wrote this I've changed my mind on how to handle unknown values, make line 47 match on the last enum value (OriginalFilenameAndId) and then make the fallthrough `_` throw a new ArgumentException saying the given value for `format` is unknown.